gbpservice/nfp/configurator/drivers/firewall/paloalto/paloalto_fw_driver.py

```diff
@@ -0,0 +1,875 @@
+        except pan.xapi.PanXapiError as err:
+            self._print_exception('PanXapiError', err, const.COMMIT)
+            raise pan.xapi.PanXapiError(err)
+        except pan.config.PanConfigError as err:
+            self._print_exception('PanConfigError', err, const.COMMIT)
+            raise pan.config.PanConfigError(err)
+        except Exception as err:
+            self._print_exception('UnexpectedError', err, const.COMMIT, resp)
+            raise Exception(err)
+
+        if self.analyze_response("COMMITED the configuration to Service VM",
+                 resp, const.COMMIT) == const.STATUS_ERROR:
+            return const.STATUS_ERROR
+
+        # wait till commit is finished
+        job_id = resp['response']['result']['job'] 
+        self.check_commit_status(vm_mgmt_ip, job_id)
+
+        return const.STATUS_SUCCESS
+
+    def apply_static_ip(self, hostname, interface_name, interface_ip):
+        LOG.info("Applying static ip '%s' to interface '%s'" % \
+                                    (interface_ip, interface_name))
+        element = (const.INTERFACE_CONFIG_TEMPLATE % \
+                                (interface_name, interface_ip))
+        try:
+            resp = self.edit_interface(hostname, interface_name, element)
+        except pan.xapi.PanXapiError as err:
+            self._print_exception('PanXapiError', err,
+                                  const.APPLY_STATIC_IP)
+            raise pan.xapi.PanXapiError(err)
+        except pan.config.PanConfigError as err:
+            self._print_exception('PanConfigError', err,
+                                  const.APPLY_STATIC_IP)
+            raise pan.config.PanConfigError(err)
+        except Exception as err:
+            self._print_exception('UnexpectedError', err,
+                                  const.APPLY_STATIC_IP, resp)
+            raise Exception(err)
+
+    def clear_interfaces(self, context, kwargs):
+        """ Clear interfaces for the service VM.
+
+        Calls static IP clear function and implements
+        persistent rule deletion in the service VM.
+        Issues REST call to service VM for deletion of interfaces.
+
+        :param context: neutron context
+        :param kwargs: a dictionary of firewall rules and objects
+        send by neutron plugin
+
+        Returns: SUCCESS/Failure message with reason.
+
+        """
+        return const.STATUS_SUCCESS
+
+    def configure_routes(self, context, kwargs):
+        """ Configure routes for the service VM.
+
+        Issues REST call to service VM for configuration of routes.
+
+        :param context: neutron context
+        :param kwargs: a dictionary of firewall rules and objects
+        send by neutron plugin
+
+        Returns: SUCCESS/Failure message with reason.
+
+        """
+        vm_mgmt_ip = kwargs.get('mgmt_ip')
+
+        # add pbf rule
+        gateway_ip = kwargs.get('gateway_ip')
+        pbf_name = const.PBF_RULE_NAME
```

Comment:
All the constants specific to palo alto should be moved to different constants file.